source/DamageProfile.cpp

```diff
@@ -115,7 +115,10 @@ void DamageProfile::PopulateDamage(DamageDealt &damage, const Ship &ship) const
 	// the value of its protection attribute.
 	auto ScaleType = [&](double blocked, double protection)
```

Comment:
Instead of having negative values somehow flip into using a different behavior, I'd prefer if this instead accepted a shield blocked and a hull blocked variable. If the shields are still up, use the shield blocked. If the shields are down, use the hull blocked.

If this change is being made to disruption, it should perhaps also be made to discharge. Discharge already currently deals 100% of its damage while the shields are up.

Reply:
Okay, I've rewritten it to take inputs "shieldBlocked", "hullBlocked", and of course "protection", which based on "shieldFraction" will return the amount you'd expect. I could make discharge damage be 50% blocked by hull (because then there's no more shields to discharge, just as disruption is halved when there are no shields to disrupt).

I originally wrote it with a secondary behavior with a negative input because I didn't see much use in, say, shieldBlocked = 50% and hullBlocked _also_ = 50%. Just go through the data files and halve that damage type across the board. But this is more clear.